test/extended/openstack/loadbalancers.go

```diff
@@ -604,35 +616,68 @@ func getFipbyFixedIP(client *gophercloud.ServiceClient, vip string) (floatingips
 	return result, fmt.Errorf("FIP not found for VIP %q", vip)
 }
 
-// Send message on provided ip and UDP port and return the answer.
-// Error if there is no answer or cannot access the port after 5 seconds.
-func getPodNameThroughUdpLb(ip string, port string, message string) (string, error) {
+func getPodNameThroughLb(ip string, port string, protocol v1.Protocol, message string) (string, error) {
 
-	service := ip + ":" + port
-	RemoteAddr, err := net.ResolveUDPAddr("udp", service)
-	if err != nil {
-		return "", err
-	}
-	conn, err := net.DialUDP("udp", nil, RemoteAddr)
-	if err != nil {
-		return "", err
-	}
-	conn.SetDeadline(time.Now().Add(time.Second * 5))
-	defer conn.Close()
+	if protocol == v1.ProtocolUDP {
+		// Send message on provided ip and UDP port and return the answer.
+		// Error if there is no answer or cannot access the port after 5 seconds.
 
-	msg := []byte(message)
-	_, err = conn.Write(msg)
-	if err != nil {
-		return "", err
-	}
+		service := ip + ":" + port
+		RemoteAddr, err := net.ResolveUDPAddr("udp", service)
+		if err != nil {
+			return "", err
+		}
+		conn, err := net.DialUDP("udp", nil, RemoteAddr)
+		if err != nil {
+			return "", err
+		}
+		conn.SetDeadline(time.Now().Add(time.Second * 5))
+		defer conn.Close()
 
-	// receive message from server
-	buffer := make([]byte, 1024)
-	n, _, err := conn.ReadFromUDP(buffer)
-	if err != nil {
-		return "", err
+		msg := []byte(message)
+		_, err = conn.Write(msg)
+		if err != nil {
+			return "", err
+		}
+
+		// receive message from server
+		buffer := make([]byte, 1024)
+		n, _, err := conn.ReadFromUDP(buffer)
```

Comment:
Since you're discarding the UPDAddr, this could be `conn.ReadFrom(buffer)`

Reply:
Done